### PR TITLE
experiment: ignore bot account ID for views

### PR DESCRIPTION
**Pull request by Zolmeister**

### CODE REVIEW

File: controllers/experiment.coffee [hunk 1]
```diff
@@ -7,14 +7,26 @@ config = require '../config'
 
 overrides = {}
 
+registerViewEvents = (isOrganic, isBot, params, userId, timestamp) ->
```
Comment (austinhallock) [inline controllers/experiment.coffee]: why not use obj notation here?

File: tests/end_to_end/result.coffee [hunk 2]
```diff
@@ -230,3 +232,42 @@ describe 'Result Routes', ->
...
+        .post '/experiments', {userId: 123}
+        .expect 200
+        .post '/experiments', {userId: 129}
+        .expect 200
+        .post '/experiments', {userId: config.CRAWLER_USER_ID}
+        .expect 200
+        .post '/conversions', {event: 'engaged_gameplay', userId: 123}
+        .expect 200
+        .get "/results?#{queryParams}"
+        .flare (x) -> console.log x.res.body
```
Comment (austinhallock) [inline tests/end_to_end/result.coffee]: rm

File: controllers/experiment.coffee [hunk 3]
```diff
@@ -7,14 +7,26 @@ config = require '../config'
...
+    unless isBot
+      log.info "VIEW: #{userId} #{JSON.stringify params}"
+      return Experiments.registerView params, userId, timestamp
+
 class ExperimentCtrl
   assign: (req) ->
-    userId = req.body.userId
-    userId ?= req.body.id # LEGACY
-    fromUserId = req.body.fromUserId
+    userId = req.body.userId and String req.body.userId
```
Comment (austinhallock) [inline controllers/experiment.coffee]: should probably use `String` in conversions.coffee too. And we should ignore bot conversions (eg a bot may stay on a page for a minute, counting as an engaged play)

File: controllers/experiment.coffee (hunk 1 above)
Comment (Zolmeister) [inline controllers/experiment.coffee]: all parameters are required, and it is not a constructor

**Comment by Zolmeister** [comment]
updated, no rebase

**Comment by Zolmeister** [comment]
updated, rebased

**Pull request merged**